tests/foreman/api/test_hostgroup.py

```diff
@@ -1,9 +1,175 @@
+            url=FAKE_0_PUPPET_REPO,
+        ).create()
+        puppet_repo.sync()
+        with open(get_data_file(PUPPET_MODULE_NTP_PUPPETLABS), 'rb') as handle:
+            puppet_repo.upload_content(handle)
+
+        content_view = entities.ContentView(
+            organization=org,
+            name=gen_string('alpha'),
+        ).create()
+        results = content_view.available_puppet_modules()['results']
+        # Working with 'ntp' module as we know for sure that it contains at
+        # least few puppet classes
+        for result in results:
+            if result['name'] == 'ntp':
+                content_view.add_puppet_module(
+                    result['author'],
+                    result['name']
+                )
+                break
+
+        content_view.publish()
+        cvv = entities.ContentViewVersion(
+            id=content_view.read_json()['versions'][0]['id']
+        )
+        lc_env = entities.LifecycleEnvironment(
+            organization=org,
+            name=gen_string('alpha')
+        ).create()
+        cvv.promote(lc_env.id)
+        cv_attrs = content_view.read_json()
+        self.assertEqual(len(cv_attrs['versions']), 1)
+        self.assertEqual(len(cv_attrs['puppet_modules']), 1)
+
+        # Form environment name variable for our test
+        env_name = 'KT_{0}_{1}_{2}_{3}'.format(
+            org.name, lc_env.name, content_view.name, str(content_view.id))
+
+        # Get all environments for current organization.
+        # We have two environments (one created after publishing and one more
+        # was created after promotion), so we need to select promoted one
+        response = client.get(
+            entities.Environment().path(),
+            auth=get_server_credentials(),
+            data={
+                u'organization_id': org.id,
+            },
+            verify=False,
+        )
+        response.raise_for_status()
+        results = response.json()['results']
+        self.assertEqual(len(results), 2)
+        env_id = None
+        for result in results:
+            if result['name'] == env_name:
+                env_id = result['id']
+                break
+        environment = entities.Environment(
+            id=env_id
+        ).read()
+
+        # Prepare necessary data for HostGroup entity
+        mac = entity_fields.MACAddressField()
+        mac = mac.gen_value()
+        root_pass = entity_fields.StringField(length=(8, 30))
+        root_pass = root_pass.gen_value()
+        domain = entities.Domain().create(True)
+        architecture = entities.Architecture().create(True)
+        ptable = entities.PartitionTable().create(True)
+        operatingsystem = entities.OperatingSystem(
+            architecture=[architecture],
+            ptable=[ptable],
+        ).create(True)
+        medium = entities.Media(
+            operatingsystem=[operatingsystem]
+        ).create(True)
+
+        host_group = entities.HostGroup(
```

Comment:
This relies on a recent change of nailgun, and AFAIK that wasn't released yet. We need to wait until the HostGroup class update is released.